plugins/mission/mission_impl.h

```diff
@@ -74,6 +79,14 @@ class MissionImpl : public PluginImplBase
     void download_next_mission_item();
     void assemble_mission_items();
 
+    static Mission::Result
+    compose_mission_items(Mission::mission_items_t &mission_items,
+                          const Json &mission_json);
+    static Mission::Result
+    compose_mission_items(MAV_CMD cmd, std::vector<double> params,
+                          std::shared_ptr<MissionItem> &new_mission_item,
+                          Mission::mission_items_t &mission_items);
```

Comment:
Two methods with different arguments? That's again confusing, I think.

Reply:
Yes, I can change as:
```
static Mission::Result
compose_mission_items(Mission::mission_items_t &mission_items, const Json &mission_json);

static Mission::Result
compose_each_mission_item(Mission::mission_items_t &mission_items, MAV_CMD cmd, std::vector<double> params, std::shared_ptr<MissionItem> &new_mission_item);
```
I could do this in a single method, but it will too big.

Reply:
No, I think the method name doesn't really explain what it does and the fact that there are two methods with the same name is confusing.